platform/security/idp/security-idp-server/src/test/groovy/org/codice/ddf/security/idp/server/IdpEndpointSpecTest.groovy

```diff
@@ -386,6 +400,59 @@ class IdpEndpointSpecTest extends Specification {
+        String relayState = "relayState"
+        def request = Mock(HttpServletRequest) {
+            getCookies() >> {
+                [Mock(Cookie) {
+                    getName() >> Idp.COOKIE
+                    getValue() >> "cookieValue"
+                }]
+            }
+        }
+
+        idpEndpoint.logoutMessage = Mock(LogoutMessage) {
+            extractSamlLogoutRequest(_ as String) >> Mock(LogoutRequest) {
+                getIssuer() >> Mock(Issuer) {
+                    getValue() >> sp1entityid
+                }
+                getNameID() >> Mock(NameID) {
+                    getValue() >> "nameId"
+                }
+                getID() >> "requestId"
+            }
+            buildLogoutResponse(_ as String, _ as String, _ as String) >>
+                    { String issuer, String statusCode, String inResponseTo ->
+                        return new LogoutResponseBuilder().buildObject()
+                    }
+        }
+
+        idpEndpoint.setSpMetadata(ImmutableList.of(sp1metadata, sp2metadata))
+
+        when:
+        def response = idpEndpoint.processPostLogout(
+                samlRequest,
+                null,
+                relayState,
+                request)
+        
+        then:
+        // Verify the request was read correctly
+        notThrown(Exception)
```

Comment:
✏️ You can skip this, as making it to the next assertion will imply that no exception was thrown.